### PR TITLE
Bump mypy to latest version

### PR DESCRIPTION
**Pull request by maffoo**
Prompted by discussion on #5486; this turned out to be <strike>not as difficult as I thought</strike> somewhat difficult.

### CODE REVIEW

**Review by mpharrigan — DISMISSED** [review]

**Review by mpharrigan — COMMENTED** [review]
type check test is failing

**Comment by maffoo** [comment]
> type check test is failing

Yeah, there seem to be differences between running mypy in my local environment and in CI, so I missed these failures locally. I'll take a look to see how tricky these errors are to fix.

File: cirq-core/cirq/sim/clifford/stabilizer_state_ch_form.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def inner_product_of_state_and_x(self, x: int) -> Union[float, complex]:
             * 2 ** (-sum(self.v) / 2)
             * 1j**mu
             * (-1) ** sum(self.v & u & self.s)
-            * np.all(self.v | (u == self.s))
+            * bool(np.all(self.v | (u == self.s)))
```
Comment (mpharrigan) [inline cirq-core/cirq/sim/clifford/stabilizer_state_ch_form.py:122]: can you use `(...).item()` to turn numpy thing into python thing?

**Comment by mpharrigan** [comment]
> there seem to be differences between running mypy in my local environment and in CI

This sounds worrying. Do you have any idea why?

**Comment by dabacon** [comment]
> rences between running mypy in my local environment and in CI, so I missed these failures locally. I'll take a look to see how

Yeah I had this too, and was a very recent install.  Maybe because our python versions are higher?

**Comment by maffoo** [comment]
I _think_ I've got everything now; PTAL @mpharrigan, @dabacon.

I found a smattering of random lint errors that were revealed when touching various files. FWIW I think we should change CI to lint everything, because the current incremental linting means there can be lint errors hiding in a file that have to be handled by the next person to touch the file for unrelated reasons.

**Review by dstrain115 — APPROVED** [review]
These look like reasonable changes to me.

**Pull request merged**